apollo-federation/src/sources/connect/json_selection/lit_expr.rs

```diff
@@ -96,7 +96,14 @@ impl LitExpr {
                         );
 
                         let mut s = String::new();
-                        s.push_str(int.fragment());
+                        let int_str = int.fragment();
+                        // Remove leading zeros to avoid failing the stricter
+                        // number.parse() below, but allow a single zero.
+                        if let Some(first_non_zero) = int_str.chars().position(|c| c != '0') {
```

Comment:
This is a super nit, but apparently `Iterator::position` [can panic](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position) if the string is long enough to overflow `usize::MAX`. I don't think that we need to handle this, but if so we should probably make use of [Iterator::skip_while](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip_while) and the like.

Reply:
https://github.com/apollographql/router/pull/6766/commits/bfccf2a3a0f8fa7ebfbc495a4c8e616ddb3443c5